zio-http/shared/src/main/scala/zio/http/StreamingForm.scala

```diff
@@ -164,24 +160,31 @@ final case class StreamingForm(source: ZStream[Any, Throwable, Byte], boundary:
 }
 
 object StreamingForm {
-  private final case class State(
```

Comment:
Why is this now a class instead of case class? Is there a benefit that I am missing?
But if we keep this a class:
This is actually some kind of iteration state, right? Would it makes sense, since we have methods like rest and so on, to have mutable fields and reassign them to avoid allocation of new `State` instances? I did not check in detail, but it seems that one state is only progressed in a linear fashion. So I don't see that this would lead to race conditions.

Reply:
@987Nabil The main reason for making it a class was so that the `copy` method is not created. The only "real" benefit from this is that prevents someone in the future assigning the old `formState` without knowing that the class mutates.

I like your suggestion though, I think if we'll use a mutable class we might as well use internal variables and avoid the extra allocations